### PR TITLE
[SPARK-10061] [DOC] ML ensemble docs

### PR DESCRIPTION
**Pull request by jkbradley**
User guide for spark.ml GBTs and Random Forests.
The examples are copied from the decision tree guide and modified to run.

I caught some issues I had somehow missed in the tree guide as well.

I have run all examples, including Java ones.  (Of course, I thought I had previously as well...)

CC: @mengxr @manishamde @yanboliang

### CODE REVIEW

**Comment by jkbradley** [comment]
Note: This is blocked by [https://github.com/apache/spark/pull/8367]

**Comment by jkbradley** [comment]
Rebased to include GBT fix in master.

**Comment by yanboliang** [comment]
Looks good!

File: docs/ml-decision-tree.md [hunk 1]
```diff
@@ -416,61 +413,49 @@ DataFrame testData = splits[1];
...
-IndexToString labelConverter = new IndexToString()
-  .setInputCol("prediction")
-  .setOutputCol("predictedLabel")
-  .setLabels(labelIndexer.labels());
-
-// Chain indexers and tree in a Pipeline
+// Chain indexer and tree in a Pipeline
 Pipeline pipeline = new Pipeline()
-  .setStages(new PipelineStage[]{labelIndexer, featureIndexer, dt, labelConverter});
+  .setStages(new PipelineStage[]{featureIndexer, dt});
```
Comment (mengxr) [inline docs/ml-decision-tree.md]: space after `[]`

File: mllib/src/main/scala/org/apache/spark/mllib/tree/configuration/Strategy.scala [hunk 2]
```diff
@@ -158,6 +158,11 @@ class Strategy (
       s"  Valid values are integers >= 0.")
     require(maxBins >= 2, s"DecisionTree Strategy given invalid maxBins parameter: $maxBins." +
       s"  Valid values are integers >= 2.")
+    categoricalFeaturesInfo.foreach { case (feature, arity) =>
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/tree/configuration/Strategy.scala]: revert changes? #8367 was merged.

**Comment by mengxr** [comment]
LGTM except some minor comments.

**Comment by jkbradley** [comment]
OK I think that does it.  Thanks for reviewing!

**Comment by mengxr** [comment]
Merged into master and branch-1.5. Thanks!

**Pull request closed without merging**